datalad/distribution/siblings.py

```diff
@@ -239,30 +246,47 @@ def __call__(
         # at the top-level and vice versa
         worker = action_worker_map[action]
 
-        dataset = require_dataset(
-            dataset, check_installed=False, purpose='configure sibling')
-        refds_path = dataset.path
+        ds = require_dataset(
+            dataset,
+            # it makes no sense to use this command without a dataset
+            check_installed=True,
+            purpose='configure sibling')
+        refds_path = ds.path
 
         res_kwargs = dict(refds=refds_path, logger=lgr)
 
-        ds_name = op.basename(dataset.path)
+        ds_name = op.basename(ds.path)
 
         # do not form single list of datasets (with recursion results) to
         # give fastest possible response, for the precise of a long-all
         # function call
-        ds = dataset
-        for r in worker(
-                # always copy signature to below to avoid bugs!
-                ds, name,
-                ds.repo.get_remotes(),
-                # for top-level dataset there is no layout questions
-                _mangle_urls(url, ds_name),
-                _mangle_urls(pushurl, ds_name),
-                fetch, description,
-                as_common_datasrc, publish_depends, publish_by_default,
-                annex_wanted, annex_required, annex_group, annex_groupwanted,
-                inherit, get_annex_info,
-                **res_kwargs):
+
+        # minimize expensive calls to .repo
+        ds_repo = ds.repo
+
+        # prepare parameterization package for all worker calls
+        worker_kwargs = dict(
```

Comment:
FWIW, I would have preferred to just identify common `worker_kwargs` and provide specific ones explicitly, instead of `.update()`ing them for the next invocation.  Rationale -- if later decided to re-use such kwargs dict for yet another worker invocation, extra care should be taken to overload all inplace modifications for subsequent invocations.  Without inplace modification
- no such care needs to be take
- makes it clear right here which options are common for top level and subdatasets

Reply:
I know that this would be better, but it would require more substantial changes. This change only wants to aid code readability -- it does not make things worse than they were before. I do not claim a beauty price for this PR. As expressed elsewhere, I am perfectly fine with ripping `siblings` apart and rewrite it from scratch. However, this would need to involve a debate about "inherit" and "delayed super" that I do not have the juice for now.

Reply:
I had in mind just this: https://github.com/datalad/datalad/pull/5924